piqe_ocp_lib/api/resources/ocp_cluster_versions.py

```diff
@@ -73,3 +74,103 @@ def get_cluster_id(self):
+            "resourceVersion": cluster_version.metadata.resourceVersion,  # Required for successfull API call
+        }
+        spec = {"clusterId": cluster_version.spec.clusterID}  # Required for successfull API call
+
+        cv_body["metadata"] = {**cv_body.get("metadata", {}), **metadata}
+        cv_body["spec"] = {**cv_body.get("spec", {}), **spec}
+        return cv_body
+
+    def available_updates(self) -> List[str]:
+        """
+        Get available versions to update
+        :return: list of available versions
+        """
+        cluster_version = self.get_cluster_version()
+        available_updates = cluster_version.items[0].status.availableUpdates
+
+        try:
+            return [element.version for element in available_updates[::-1]]
+        except TypeError:
+            logger.debug("There's no update available.")
+
+        return []
+
+    def available_channels(self) -> List[str]:
+        """
+        """
+        cluster_version = self.get_cluster_version()
+        available_channels = cluster_version.items[0].status.availableUpdates
+        unique_channels = set().union(*[e.channels for e in available_channels])
+        return list(unique_channels) or []
+
+    def latest_version_available(self) -> str:
+        """
+        Get latest version available to upgrade
+        :return: String with latest value OR None
+        """
+        cluster_available_updates = self.available_updates()
+
+        try:
+          return cluster_available_updates[-1]
+        except IndexError:
+          logger.debug("Cluster already at latest version.")
+
+    def update_channel(self, channel: str):
+        """
+        Change update channel
+        :param channel: (str) name of the channel
+        :return: (ClusterVersion)
+        """
+        cv_body = {"spec": {"channel": channel}}
+        cv_body = self._build_spec(cv_body)
+        return self.update_cluster_version(cv_body)
+
+    def upgrade_cluster_version(self, version: str, force: bool = False, timeout: int = 0) -> Dict:
+        """
+        Upgrade cluster to desired version
+        :param force: whether to force the update or not
+        :param timeout: minutes to wait for upgrade completion
+        :return: ClusterVersion API response
+        """
+        def wait_until_upgraded(timeout: int):
+            """
+            """
+            pooling_interval = 60 * 2
+            ends_at = datetime.datetime.now() + datetime.timedelta(minutes=timeout)
+            condition = lambda oc_cv: oc_cv.items[0].status.desired.version == version
+
+            while datetime.datetime.now() < ends_at:
+                cluster_version = self.get_cluster_version()
+
+                if condition(cluster_version):
+                    logger.info("Cluster upgraded successfully.")
+                    return cluster_version
+                else:
+                    logger.info(f"Upgrade in progress, checking again in {pooling_interval} minutes.")
```

Comment:
Add status.conditions[latest_item].message for more information on what's happening